### PR TITLE
GH-2851: Git over SSH.

### PR DESCRIPTION
**Pull request by kittaakos**
Closes: #2851.

Signed-off-by: Akos Kitta <kittaakos@typefox.io>

### CODE REVIEW

File: packages/git/src/node/dugite-git.ts [hunk 1]
```diff
@@ -605,6 +627,16 @@ export class DugiteGit implements Git {
         }
     }
 
+    protected async options(): Promise<IGitExecutionOptions> {
```
Comment (AlexTugarev) [inline packages/git/src/node/dugite-git.ts]: is this unused?

Comment (kittaakos) [reply packages/git/src/node/dugite-git.ts]: 👍 Yes.

**Comment by kittaakos** [comment]
I have just noticed the failing test case on both CIs.
```
1 failing
  1) git ls-files errorUnmatched - ../missing.txt should be rejected:
      AssertionError: Unknown error: 33.: expected false to equal true
      + expected - actual
      -false
      +true
      at Suite.<anonymous> (src/node/dugite-git.spec.ts:765:74)
      at step (src/node/dugite-git.spec.ts:47:23)
      at Object.throw (src/node/dugite-git.spec.ts:28:53)
      at rejected (src/node/dugite-git.spec.ts:20:65)
      at <anonymous>
```

I fix it right now.

**Comment by kittaakos** [comment]
It should be OK, @AlexTugarev. Thanks for your patience!

**Comment by AlexTugarev** [comment]
thanks! retesting right now.

**Comment by kittaakos** [comment]
The Git tests are passing on Windows, the current test failures are unrelated.
```

2174  Preference Service
2175    1) Should get notified if a provider gets a change
2176    2) Should return the preference from the more specific scope (user > workspace)
2177    3) Should return the preference from the less specific scope if the value is removed from the more specific one
2178    4) Should throw a TypeError if the preference (reference object) is modified
2179    5) Should still report the more specific preference even though the less specific one changed
2180    √ Should store preference when settings file is empty (47ms)
2181    √ Should store preference when settings file is not empty
2182    √ Should override existing preference
2183    √ Should be ready only when all providers are ready (1015ms)
```

I am merging this.

**Pull request merged**